core/backend/src/StashManager.ts

```diff
@@ -0,0 +1,458 @@
+      throw new StashError("Invalid stash");
+    }
+    return stashFile;
+  }
+
+  /**
+   * Queries the stash database for lock IDs matching the specified state and origin.
+   *
+   * @param args - The arguments required to access the stash database.
+   * @param state - The lock state to filter by.
+   * @param origin - The lock origin to filter by.
+   * @returns An array of lock IDs (`Id64Array`) that match the given state and origin.
+   */
+  private static queryLocks(args: StashArgs, state: LockState, origin: LockOrigin): Id64Array {
+    return this.withStash(args, (stashDb) => {
+      const query = `SELECT JSON_GROUP_ARRAY(FORMAT('0x%x', Id)) FROM [locks] WHERE State = ${state} AND origin = ${origin}`;
+      return stashDb.withPreparedSqliteStatement(query, (stmt) => {
+        while (stmt.step() === DbResult.BE_SQLITE_ROW) {
+          return JSON.parse(stmt.getValueString(0)) as Id64Array;
+        }
+        return [];
+      });
+    });
+  }
+
+  /**
+   * Acquire locks for the specified stash. If this fail then stash should not be applied.
+   * @param args The stash arguments.
+   */
+  private static async acquireLocks(args: StashArgs) {
+    const shared = this.queryLocks(args, LockState.Shared, LockOrigin.Acquired);
+    await args.db.locks.acquireLocks({ shared });
+
+    const exclusive = this.queryLocks(args, LockState.Exclusive, LockOrigin.Acquired);
+    await args.db.locks.acquireLocks({ exclusive });
+
+    const newElements = this.queryLocks(args, LockState.Shared, LockOrigin.NewElement);
+    for (const id of newElements) {
+      if (!args.db.locks.holdsExclusiveLock(id)) {
+        args.db.locks[_elementWasCreated](id);
+      }
+    }
+  }
+  /**
+   * Creates a stash of changes for the specified briefcase.
+   *
+   * This method generates a stash in the stash root directory for the given briefcase, using the provided description and iModelId.
+   * Optionally, it can reset the briefcase by releasing all locks after stashing.
+   *
+   * @param args - The properties required to create a stash, including the briefcase, description, iModelId, and an optional resetBriefcase flag.
+   * @returns A promise that resolves to the properties of the created stash.
+   *
+   * @throws Error if the stash operation fails.
+   */
+  public static async stash(args: CreateStashProps): Promise<StashProps> {
+    const stashRootDir = StashManager.getStashRootFolder(args.db, true);
+    const iModelId = args.db.iModelId;
+    if (!args.db.txns.hasPendingTxns) {
+      throw new StashError("nothing to stash");
+    }
+
+    if (args.db.txns.hasUnsavedChanges) {
+      throw new StashError("Unsaved changes exist");
+    }
+
+    if (args.db.txns.hasPendingSchemaChanges) {
+      throw new StashError("Pending schema changeset. Stashing is not currently supported for schema changes");
+    }
+
+    const stash = args.db[_nativeDb].stashChanges({ stashRootDir, description: args.description, iModelId }) as StashProps;
```

Comment:
Is there a reason that `stashChanges` returns `any` instead of `StashProps`? Is that to hide `StashProps` from end users?